pandas/core/indexes/base.py

```diff
@@ -359,7 +360,10 @@ def _outer_indexer(
 
     _typ: str = "index"
     _data: ExtensionArray | np.ndarray
-    _data_cls: type[np.ndarray] | type[ExtensionArray] = np.ndarray
+    _data_cls: type[ExtensionArray] | tuple[type[np.ndarray], type[ExtensionArray]] = (
+        np.ndarray,
+        ExtensionArray,
```

Comment:
ended up adding a comment down where we use _data_cls, but agreed this isn't pretty.

Reply:
But what's the point of defining it here as a tuple, if that is actually never used? (or for future use?)

Because for keeping the attribute for sublcasses, it could also simply be an optional EA ? (with a comment this is only used for subclasses that actually use an EA) 

```
_data_cls: type[ExtensionArray] | None = None
```

Reply:
in Index._simple_new we have `assert isinstance(values, cls._data_cls)`

Reply:
Ah, missed that one usage.